src/frontend/components/Chat/SharedChatComponents/ChatPresenceItem/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { Box, Text } from 'grommet';
+import React from 'react';
+
+import {
+  ChatPresenceType,
+  presenceType,
+} from 'data/stores/usePresenceItemsStore';
+import { defineMessages, useIntl } from 'react-intl';
+
+interface ChatPresenceItemProps {
+  presenceItem: ChatPresenceType;
+}
+
+const messages = defineMessages({
+  joinedChatLabel: {
+    defaultMessage: 'has joined',
+    description: 'A label to announce that a user has joined the chat.',
+    id: 'component.ChatPresenceItem.joinedChatLabel',
+  },
+  leftChatLabel: {
+    defaultMessage: 'has left',
+    description: 'A label to announce that a user has left the chat.',
+    id: 'component.ChatPresenceItem.leftChatLabel',
+  },
+});
+
+export const ChatPresenceItem = ({ presenceItem }: ChatPresenceItemProps) => {
```

Comment:
you should maybe split this in 2 components (like ChatPresenceItem and ChatPresenceItemLayout), the first one computing the text to display, the second displaying it with all the style

I'm not 100% sure in this case because the computation is very simple, yet this pattern is quite simple to use and split very simply the code isolating components